### PR TITLE
Super pipeline generator

### PR DESCRIPTION
**Pull request by Mohammad-nassar10**
## Why are these changes needed?
Add a tool that helps to generate super pipelines python code.

### CODE REVIEW

File: kfp/pipeline_generator/superpipeline/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## Steps to generate a new pipeline
```
Comment (revit13) [inline kfp/pipeline_generator/superpipeline/README.md]: Maybe mention it support only kfp v1. Also, you can add a link to data-prep-lab/kfp/doc/multi_transform_pipeline.md for more info on super pipeline,,,

Comment (Mohammad-nassar10) [reply kfp/pipeline_generator/superpipeline/README.md]: Fixed.

**Comment by blublinsky** [comment]
@Mohammad-nassar10, @revit13, @roytman is this still relevant? It has been dormant for a while

**Comment by Mohammad-nassar10** [comment]
@blublinsky yes, I updated in the past two days.

**Review by revit13 — APPROVED** [review]

File: kfp/pipeline_generator/superpipeline/README.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+## Steps to generate a new super pipeline in KFP v1.
+- The super pipeline is a way to execute several transforms with one pipeline. You can find more details in [multi_transform_pipeline.md](../../doc/multi_transform_pipeline.md)
```
Comment (roytman) [inline kfp/pipeline_generator/superpipeline/README.md]: Please restrict line length to 80 characters:

How about the following  explanation ?
- The super pipeline allows you to execute several transforms within a single pipeline. For more details, refer to [multi_transform_pipeline.md](../../doc/multi_transform_pipeline.md)
- Create a super_pipeline_definitions.yaml file for the required task. You can refer to the example [super_pipeline_definitions.yaml](./super_pipeline_definitions.yaml).
- Execute ./run.sh < super_pipeline_definitions.yaml> <destination_directory>. Here, `super_pipeline_definitions.yaml` is the super pipeline definition file, that you created above, and destination_directory is the directory where the new super pipeline file will be generated.

*__Note__*: The `component_spec_path` is the path to the `kfp_ray_components` folder and depends on where the workflow is compiled.

File: kfp/pipeline_generator/superpipeline/run.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+DEF_FILE=$1
```
Comment (roytman) [inline kfp/pipeline_generator/superpipeline/run.sh]: can you take examples of arguments processing from `run.sh` in the  single pipeline generator

File: kfp/pipeline_generator/superpipeline/README.md (hunk 2 above)
Comment (roytman) [reply kfp/pipeline_generator/superpipeline/README.md]: where do we set `component_spec_path` ?

Comment (Mohammad-nassar10) [reply kfp/pipeline_generator/superpipeline/README.md]: The Readme.md updated.
The user can set the `component_spec_path` in the `super_pipeline_definitions.yaml`.

File: kfp/pipeline_generator/superpipeline/run.sh (hunk 3 above)
Comment (Mohammad-nassar10) [reply kfp/pipeline_generator/superpipeline/run.sh]: I changed the implementation to use jinja as in the single pipeline generator.

**Review by roytman — COMMENTED** [review]
LGTM

**Pull request merged**